test/test_cuda_sanitizer.py

```diff
@@ -491,6 +492,20 @@ def test_error_message(self):
             ),
         )
 
+    def test_subclass(self):
+        class MyT(torch.Tensor):
+            def __new__(cls, data):
+                new_data = data.clone()
+                return new_data.as_subclass(cls)
+
+        try:
+            csan.enable_cuda_sanitizer()
+            t = TwoTensor(torch.rand(2), torch.rand(2))
+
+            t = MyT(torch.rand(2))
```

Comment:
I don't understand how this test case would have failed before. It looks like we expected a race for creating two different subclasses ?

Reply:
No just two different kind of subclass should work.